README.md

```diff
@@ -0,0 +1,131 @@
+# Dropshare Connection - AWS S3
+
+> Terraform Module to create resources for a Dropshare Connection with AWS S3 and AWS CloudFront.
+
+## Table of Contents
+
+- [Dropshare Connection: AWS S3](#dropshare-connection-aws-s3)
+  - [Table of Contents](#table-of-contents)
+  - [Overview](#overview)
+  - [Usage](#usage)
+    - [Dropshare](#dropshare)
+    - [Inputs](#inputs)
+    - [Outputs](#outputs)
+  - [Notes](#notes)
+  - [Author Information](#author-information)
+  - [License](#license)
+
+## Overview
+
+```mermaid
+  graph TD;
+    dns_a[DNS Record, type A] --> tls;
+    dns_aaaa[DNS Record, type AAAA] --> tls;
+    tls[TLS Certificate] --> cloudfront;
+
+    cloudfront[CloudFront Distribution];
+
+    bucket[S3 Bucket] --> cloudfront;
+
+    user[IAM User] --> bucket;
+```
+
+## Usage
+
+This section contains the input and output values of this module as well as information on how to configure [Dropshare](https://dropshare.app/).
+
+### Dropshare
+
+To configure Dropshare, start from `basic` example available in [./examples/basic](https://github.com/ksatirli/terraform-awscc-dropshare/blob/main/examples/basic).
+
+Modify the input values `bucket_name`, `keybase_user`, `routed53_zone_id` and `route53_record_subdomain` to reflect your infrastructure.
+
+Next, carry out a [Terraform run](https://www.terraform.io/cli/run), executing the `terraform plan`, and `terraform apply`.
+
+When the [apply](https://www.terraform.io/cli/run#applying) process finishes, Terraform will present the required [outputs](https://www.terraform.io/language/values/outputs) to configure Dropshare:
+
+```shell
+Outputs:
+
+access_key_id              = "AKIAIOSFODNN7EXAMPLE"
+bucket_name                = "aws-ia-dropshare"
+domain_alias               = "aws-ia-dropshare.svcs.dev"
+region                     = "us-west-1"
+secret_key                 = "wcB...A=="
```

Comment:
perhaps an inline comment with a link to the dropshare docs for how to generate this secret key?

Reply:
The secret key is generated through this module and then encrypted with a Keybase key and made available through an output.